include/envoy/stream_info/stream_info.h

```diff
@@ -276,16 +276,30 @@ class StreamInfo {
    */
   virtual const Network::Address::InstanceConstSharedPtr& downstreamLocalAddress() const PURE;
 
+  /**
+   * @param downstream_directly_connected_address sets the direct physical address of downstream
+   * connection.
+   */
+  virtual void setDownstreamDirectRemoteAddress(
+      const Network::Address::InstanceConstSharedPtr& downstream_directly_connected_address) PURE;
```

Comment:
The variable name needs to change to something like `downstream_direct_remote_address`.